controllers/kserve_ray_tls_controller_test.go

```diff
@@ -0,0 +1,285 @@
+			Expect(err).NotTo(HaveOccurred())
+			rayCaCertUpdatedSecret.SetNamespace(WorkingNamespace)
+			Expect(cli.Update(ctx, rayCaCertUpdatedSecret)).Should(Succeed())
+
+			_, err = waitForSecret(cli, WorkingNamespace, constants.RayCASecretName, 30, 1*time.Second)
+			Expect(err).NotTo(HaveOccurred())
+
+			// Check if 'ray-ca-cert' Secret in the testNs is updated.
+			Eventually(func() bool {
+				updatedSecretFromTestNs, err := waitForSecret(cli, testNs, constants.RayCASecretName, 1, 1*time.Second)
+				Expect(err).NotTo(HaveOccurred())
+				return compareSecrets(rayCaCertUpdatedSecret, updatedSecretFromTestNs)
+			}, timeout, interval).Should(BeTrue())
+		})
+	})
+	Context("when a multinode ServingRuntime removed from the testNs", func() {
+		var testNs string
+		BeforeEach(func() {
+			testNamespace := Namespaces.Create(cli)
+			testNs = testNamespace.Name
+
+			// Create ray tls script in the ctrl namespace
+			rayTlsScriptsConfigMap := &corev1.ConfigMap{}
+			err := convertToStructuredResource(rayTlsScriptsPath, rayTlsScriptsConfigMap)
+			Expect(err).NotTo(HaveOccurred())
+			rayTlsScriptsConfigMap.SetNamespace(WorkingNamespace)
+			Expect(cli.Create(ctx, rayTlsScriptsConfigMap)).Should(Succeed())
+
+			// Create a multinode ServingRuntime in the testNs
+			multinodeServingRuntime := &kservev1alpha1.ServingRuntime{}
+			err = convertToStructuredResource(multinodeServingRuntimePath, multinodeServingRuntime)
+			Expect(err).NotTo(HaveOccurred())
+			multinodeServingRuntime.SetNamespace(testNs)
+			Expect(cli.Create(ctx, multinodeServingRuntime)).Should(Succeed())
+
+			_, err = waitForConfigMap(cli, testNs, constants.RayTlsScriptConfigMapName, 30, 1*time.Second)
+			Expect(err).NotTo(HaveOccurred())
+			_, err = waitForSecret(cli, testNs, constants.RayCASecretName, 30, 1*time.Second)
+			Expect(err).NotTo(HaveOccurred())
+		})
+		It("ray tls script should not be removed if there is a multinode ServingRuntime in the testNs", func() {
+			By("creating another multinode ServingRuntime in the testNs")
+			// Create another multinode ServingRuntime
+			multinodeServingRuntime := &kservev1alpha1.ServingRuntime{}
+			err := convertToStructuredResource(multinodeServingRuntimePath, multinodeServingRuntime)
+			Expect(err).NotTo(HaveOccurred())
+			multinodeServingRuntime.SetNamespace(testNs)
+			multinodeServingRuntime.SetName("another-multinode-servingruntime")
+			Expect(cli.Create(ctx, multinodeServingRuntime)).Should(Succeed())
+
+			By("deleting one multinode ServingRuntime in the testNs")
+			Expect(cli.Delete(ctx, multinodeServingRuntime)).Should(Succeed())
+
+			// Check if all ray tls script are NOT removed
+			_, err = waitForConfigMap(cli, testNs, constants.RayTlsScriptConfigMapName, 30, 1*time.Second)
+			Expect(err).NotTo(HaveOccurred())
+			_, err = waitForSecret(cli, testNs, constants.RayCASecretName, 30, 1*time.Second)
+			Expect(err).NotTo(HaveOccurred())
```

Comment:
It is better to surround these inside `Consistently` to make sure the conditions remain true over a timespan.